cpp/test/prims/make_arima.cu

```diff
@@ -70,16 +67,11 @@ class MakeArimaTest : public ::testing::TestWithParam<MakeArimaInputs> {
                params.gtype);
   }
 
-  void TearDown() override
-  {
-    CUDA_CHECK(cudaFree(data));
-    CUDA_CHECK(cudaStreamDestroy(stream));
-  }
+  void TearDown() override { CUDA_CHECK(cudaStreamDestroy(stream)); }
 
  protected:
   MakeArimaInputs params;
-  T* data;
-  std::shared_ptr<raft::mr::device::allocator> allocator;
+  std::unique_ptr<rmm::device_uvector<T>> data;
```

Comment:
I won't comment on all these, but we should review all the unique_ptrs to device_uvector and see if they're actually useful.